drv/stm32h7-qspi/src/lib.rs

```diff
@@ -186,8 +186,19 @@ impl Qspi {
         // off the front.
         let mut data = data;
         while !data.is_empty() {
+            let sr = self.reg.sr.read();
+
+            if sr.tof().bit_is_set() {
+                self.reg.fcr.modify(|_, w| w.ctof().set_bit());
```

Comment:
Do you need to clear interrupt flags here (and below) before returning the error?

Reply:
I think I lost that in one of my changes but I also see a few more errors I may need to fix :upside_down_face: